src/PromiseType.cc

```diff
@@ -51,30 +55,33 @@ static bool onResolvedCb(JSContext *cx, unsigned argc, JS::Value *vp) {
   PyObject *futureObj = (PyObject *)(futureObjVal.toPrivate());
 
   // Settle the Python asyncio.Future by the Promise's result
-  PyEventLoop::Future future = PyEventLoop::Future(futureObj);
+  PyEventLoop::Future future = PyEventLoop::Future(futureObj); // will decrease the reference count of `futureObj` in its destructor when the `onResolvedCb` function ends
   if (state == JS::PromiseState::Fulfilled) {
     future.setResult(result);
   } else { // state == JS::PromiseState::Rejected
     future.setException(result);
   }
 
   Py_DECREF(result);
+  // Py_DECREF(futureObj) // the destructor for the `PyEventLoop::Future` above already does this
```

Comment:
Can we remove this permanently?
```suggestion
```

Reply:
No, because this comment would tell later developers on this why we don't need a `Py_DECREF(futureObj)` here.

Reply:
For sure. Still I wanted to see if that was indeed your viewpoint